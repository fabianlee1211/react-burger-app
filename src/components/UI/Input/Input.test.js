import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

describe('<Input />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Input />);
  });

  it('renders Input correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a label', () => {
    wrapper.setProps({ label: 'Testing Label' });
    expect(wrapper.find('label.Label')).toHaveLength(1);
    expect(wrapper.find('label.Label').text()).toBe('Testing Label');
  });

  it('renders different input types', () => {
    const elementTypes = ['input', 'textarea', 'select'];
    const options = [{ value: 'test', displayValue: 'Test' }];
    const wrapper = shallow(<Input elementConfig={{ options }} />);
    elementTypes.forEach(element => {
      wrapper.setProps({ elementType: element });
      expect(wrapper.find(element)).toHaveLength(1);
      if (element === 'select') expect(wrapper.find('option')).toHaveLength(1);
    });
  });

  it('renders error message when invalid', () => {
    const wrapper = shallow(<Input invalid touched shouldValidate />);
    expect(wrapper.find('p.ErrorMessage')).toHaveLength(1);
  });

  it('can call onChange', () => {
    const onChange = jest.fn();
    wrapper.setProps({ changed: onChange });
    wrapper.find('input').simulate('change', 'Test');
    expect(onChange).toHaveBeenCalledWith('Test');
  });
});
