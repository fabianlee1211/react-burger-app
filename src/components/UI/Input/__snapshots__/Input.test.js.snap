// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders Input correctly 1`] = `
<div
  className="Input"
>
  <label
    className="Label"
  />
  <input
    className="InputElement"
  />
</div>
`;
